ical/event.py

```diff
@@ -93,13 +95,13 @@ class Event(ComponentModel):
     """A globally unique identifier for the event."""
 
     # Has an alias of 'start'
-    dtstart: Union[datetime.datetime, datetime.date] = Field(
+    dtstart: Union[datetime.date, datetime.datetime, None] = Field(
```

Comment:
I think what is happening here is 99% of the time want to make `dtstart` a required field, however there is a case for doing a partial update where it can be ommitted and so this is a hack.  Can we avoid introducing this breaking typing change and leave it in the weird state?